components/apimgt/org.wso2.carbon.apimgt.common.gateway/src/main/java/org/wso2/carbon/apimgt/common/gateway/analytics/collectors/impl/SuccessRequestDataCollector.java

```diff
@@ -68,6 +70,11 @@ public void collectData() {
         Latencies latencies = provider.getLatencies();
         MetaInfo metaInfo = provider.getMetaInfo();
         String userAgent = provider.getUserAgentHeader();
+        String userIp = provider.getEndUserIP();
+        if (userIp == null) {
```

Comment:
Can it be empty?

Reply:
This is just to make sure send a default value of the userIP is not found. Because without userIP we can proceed not failed schema validation we set a default value. Base on the userIP identification and the way it provides this can be null. In the synapse, I don't think this will be null.